daml-script/runner/src/main/scala/com/digitalasset/daml/lf/engine/script/TestMain.scala

```diff
@@ -123,29 +122,46 @@ object TestMain extends StrictLogging {
           success = new AtomicBoolean(true)
           _ <- Future.sequence {
             dar.main._2.modules.flatMap {
-              case (moduleName, module) =>
+              case (moduleName, module) => {
+                object ScriptDefinition {
```

Comment:
Maybe worth factoring this out to a top-level definition? I’m also not a huge fan of `unapply`, I always find it a bit hart to read and iirc it also breaks completeness checks. Can we just separate this in two steps: First use `collect` as a `mapMaybe` where the function is something like the current `unapply` (you can turn a function returning an optional into a partial function using `Function.unlift` I believe or just define it directly as a partial function). Then we have a list of all scripts. And then in a second step run them.

Reply:
`Function.unlift` is a good hint, thanks! It turns out that making `Script.fromIdentifier|Dar` pure allows to simplify this bit. I've pulled it out into a separate step that collects all test scripts that take no arguments into a list using `collect(Function.unlift ...)`